time/src/date.rs

```diff
@@ -641,6 +717,110 @@ impl Date {
             None
         }
     }
+
+    /// Calculates the first occurrence of a weekday that is strictly later than a given `Date`.
+    /// Returns `None` if an overflow occurred.
+    ///
+    /// # Examples
+    /// ```
+    /// # use time::Weekday;
+    /// # use time_macros::date;
+    /// assert_eq!(date!(2023 - 06 - 28).checked_next_occurrence(Weekday::Monday), Some(date!(2023 - 07 - 03)));
+    /// assert_eq!(date!(2023 - 06 - 19).checked_next_occurrence(Weekday::Monday), Some(date!(2023 - 06 - 26)));
+    /// ```
+    pub const fn checked_next_occurrence(self, weekday: Weekday) -> Option<Self> {
```

Comment:
Let's have the checked methods `pub(crate)` for the time being. They can be made public if there's demand.

Reply:
In my latest commit, this has already been fixed.